app/models/commentaire.rb

```diff
@@ -92,11 +101,17 @@ def notify
     # - If a user or an invited user posted a commentaire, do nothing,
     #   the notification system will properly
     # - Otherwise, a instructeur posted a commentaire, we need to notify the user
-    if sent_by_instructeur? || sent_by_expert?
+    if sent_by_instructeur?
+      notify_user_with_delay
+    elsif sent_by_expert?
       notify_user
     end
   end
 
+  def notify_user_with_delay
+    NotifyNewAnswerWithDelayJob.set(wait: 5.minutes).perform_later(dossier, body, self)
```

Comment:
Il est possible d'ajouter un wait directement à un mailer (et je changerai bien la signature pour passer juste le commentaire au mailer). Et pour le check si le message est supprimé on peut le faire dans un `before_action` du mailer et désactiver l'envoi au besoin : https://github.com/betagouv/demarches-simplifiees.fr/blob/a46dd59f3f2a565da9aa1e4add87dc6424f813f6/app/mailers/notification_mailer.rb#L56

```suggestion
    DossierMailer.notify_new_answer(self).deliver_later(wait: 5.minutes)
```

Reply:
c'etait une 1ere piste, mais ne connaissant pas le `after_action` / `perform_deliveries=false` c'etait pas jolie. tx!